File: src/main/java/com/gcu/MessageService/model/MessageModel.java
package com.gcu.MessageService.model;

import com.gcu.MessageService.data.entity.MessageEntity;

import java.io.Serializable;

public class MessageModel implements Serializable {

    private String ID;
    private String message;
    private String timestamp;
    private String userID;
    private String conversationID;

    public MessageModel(){
        ID = "";
        message = "";
        timestamp = "";
        userID = "";
        conversationID = "";
    }

    public MessageModel(String ID, String message, String timestamp, String userID, String conversationID) {
        this.ID = ID;
        this.message = message;
        this.timestamp = timestamp;
        this.userID = userID;
        this.conversationID = conversationID;
    }

    public MessageModel(MessageEntity message){
        ID = message.getId();
        this.message = message.getMessage();
        timestamp = message.getTimestamp();
        userID = message.getUserID();
        conversationID = message.getConversationID();
    }

    public String getID() {
        return ID;
    }

    public void setID(String ID) {
        this.ID = ID;
    }

    public String getMessage() {
        return message;
    }

    public void setMessage(String message) {
        this.message = message;
    }

    public String getTimestamp() {
        return timestamp;
    }

    public void setTimestamp(String timestamp) {
        this.timestamp = timestamp;
    }

    public String getUserID() {
        return userID;
    }

    public void setUserID(String userID) {
        this.userID = userID;
    }

    public String getConversationID() {
        return conversationID;
    }

    public void setConversationID(String conversationID) {
        this.conversationID = conversationID;
    }

    public MessageEntity toEntity(){
        return new MessageEntity(ID, message, timestamp, userID, conversationID);
    }
}

File: src/main/kotlin/com/gcu/messageservice/model/MessageModel.kt
package com.gcu.messageservice.model

import com.gcu.messageservice.data.entity.MessageEntity

data class MessageModel(
    var ID: String,
    var message: String,
    var timestamp: String,
    var userID: String,
    var conversationID: String
){
    constructor() : this(
        "",
        "",
        "",
        "",
        ""
    )
    constructor(message: MessageEntity) : this(
        message.id,
        message.message,
        message.timestamp,
        message.userID,
        message.conversationID
    )

    fun toEntity(): MessageEntity = MessageEntity(ID, message, timestamp, userID, conversationID)
}
